test/neo4clj/client_test.clj

```diff
@@ -0,0 +1,207 @@
+  (t/testing "Run write cypher in a Neo4j read-only connection"
+    (test-utils/with-db conn {}
+      (t/is (thrown? java.lang.Exception
+                     (sut/execute-read conn "CREATE (n:TestNode) RETURN n"))))))
+
+(t/deftest execute-write!
+  "This test also tests the with-write-conn function"
+  (t/testing "Run write cypher without params in a Neo4j write connection"
+    (test-utils/with-db conn {}
+      (t/is (= (sut/execute-write! conn "CREATE (n:TestNode) RETURN n")
+               [{"n" {:labels [:test-node] :id 0 :ref-id "n" :props {}}}]))))
+  (t/testing "Run write cypher with params in a Neo4j write connection"
+    (test-utils/with-db conn {}
+      (t/is (= (sut/execute-write! conn "CREATE (n:TestNode {name: $name}) RETURN n" {:name "Neo"})
+               [{"n" {:labels [:test-node] :id 0 :ref-id "n" :props {:name "Neo"}}}])))))
+
+(t/deftest create-index!
+  (t/testing "Create index on a Neo4j node property"
+    (test-utils/with-db conn {:initial-data ["CREATE (n:TestNode {name: 'Neo'}) RETURN n"]}
+      (t/is (= (map #(select-keys % ["properties" "labelsOrTypes"])
+                    (do (sut/create-index! conn :test-node [:name])
+                        (sut/execute! conn "CALL db.indexes")))
+               [{"properties" ["name"] "labelsOrTypes" ["TestNode"]}])))))
+
+(t/deftest drop-index!
+  (t/testing "Create index on a Neo4j node property"
+    (test-utils/with-db conn {:initial-data ["CREATE INDEX ON :TestNode(name)"]}
+      (t/is (= (do (sut/drop-index! conn :test-node [:name])
+                   (sut/execute! conn "CALL db.indexes")))
+            []))))
+
+(t/deftest create-node!
+  (t/testing "Create a Neo4j node though builder"
+    (test-utils/with-db conn {}
+      (t/is (= (sut/create-node! conn {:ref-id "n" :labels [:test-node] :props {:name "Neo"}})
+               {:labels [:test-node] :id 0 :ref-id "n" :props {:name "Neo"}})))))
+
+(t/deftest create-rel!
+  (t/testing "Create a Neo4j relation though builder"
+    (test-utils/with-db conn {:initial-data ["CREATE (n:TestNode {name: 'Neo'})"
+                                             "CREATE (n:TestNode {name: 'Trinity'})"]}
+      (t/is (= (sut/create-rel! conn {:ref-id "r"
+                                      :type :friends
+                                      :from {:labels [:test-node] :props {:name "Neo"}}
+                                      :to {:labels [:test-node] :props {:name "Trinity"}}
+                                      :props {:since "1999"}})
+               {:end-id 1 :type :friends :start-id 0 :id 0 :ref-id "r" :props {:since "1999"}})))))
+
+(t/deftest find-node
+  (t/testing "Find a Neo4j node"
+    (test-utils/with-db conn {:initial-data ["CREATE (n:TestNode {name: 'Neo'})"]}
+      (t/is (= (sut/find-node conn {:ref-id "n" :labels [:test-node] :props {:name "Neo"}})
+               {:labels [:test-node] :id 0 :ref-id "n" :props {:name "Neo"}})))))
```

Comment:
Maybe add a negative test as well, searching for "Trinity" and finding nothing.

A negative test is also relevant for `find-rel`.